ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -751,11 +748,28 @@ void wasSuccessful() {
       if (status == TxnStatus.UNKNOWN) {
         return;
```

Comment:
We should probably shut down the heartbeater in this case too

Reply:
Good idea. I've refactored the `close()` method accordingly